fix two meter state issues

huss opened the pull request:
# Description

- When you edit a meter and make certain changes, it now refreshes readings so they will graph. The code to refresh now check that either refresh readings or cik occurred to avoid a window reload when it is not needed.
- The CSV readings upload now does a window refresh so any newly created meter is visible. There are TODOs in the code because it should only happen when the meter is created and it should only update the meter state rather than reload. This is being delayed until the page is converted to React hooks.

Note that you cannot change the meter unit correctly until PR #828 is merged. I put that code fix into my code to see it works correctly but did not include in this PR.

Fixes #837

## Type of change

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations

The ones stated in the description.

huss committed:
fix two meter state issues

- When you edit a meter and make certain changes, it now refreshes
readings so they will graph. The code to refresh now check that either
refresh readings or cik occurred to avoid a window reload when
it is not needed.
- The CSV readings upload now does a window refresh so any newly
created meter is visible. There are TODOs in the code because it
should only happen when the meter is created and it should only
update the meter state rather than reload. This is being delayed until
the page is converted to React hooks.

spearec commented on src/client/app/components/meters/EditMeterModalComponent.tsx:
```diff
@@ -231,8 +236,19 @@ export default function EditMeterModalComponent(props: EditMeterModalComponentPr
+				// or the unit change to/from quantity.
+				const shouldRefreshReadingViews = (props.meter.unitId != state.unitId) &&
+					((props.meter.unitId == -99 || state.unitId == -99) ||
+						(unitState[props.meter.unitId].unitRepresent == UnitRepresentType.quantity
+							&& unitState[state.unitId].unitRepresent != UnitRepresentType.quantity) ||
+						(unitState[props.meter.unitId].unitRepresent != UnitRepresentType.quantity
+							&& unitState[state.unitId].unitRepresent == UnitRepresentType.quantity));
```
This logic seems fairly complex, but after trying to simplify it a bit I found no shorter ways to do it. The comment is descriptive though so I think it is fine.

huss replied:
Yes, I wanted to make it simpler but also found that all the tests were needed.

spearec reviewed (approved):
Changes look good. The one thing I was unsure about is that there is no change to CreateMeterModalComponent - does that already properly reload?

huss commented:
Creating/editing a meter does not create a new type (only uses an existing type) and that is why you don't need to do a Cik action for either. When you edit you may need to refresh the reading views because they change. When you create a meter, there is no way to load in data so refreshing the readings would not change anything. That does need to happen if you then load via CSV or a usual meter data update (and I hope that does happen in that code).

spearec commented:
Yep, makes sense - misunderstood exactly what was going on. In that case, this PR should cove everything it needs to.

huss committed:
Merge remote-tracking branch 'origin/development' into pr/huss/902

huss committed:
fix bug in returned meters info

The newer routes return the meter info on edit and create. They were not
formatting the response so a couple of the properties had the wrong
name. This caused a bug where after creating a meter you could not
edit it until the web browser was refreshed. This should resolve this.

huss commented:
I just added another fix to this PR. PR #828 had meter creation and edit route back the updated meter info. This did not format the response so properties with names that differed from the DB name were not properly set. This meant you could not edit a meter after creating it.

I have also merged in development to resolve the merge conflict.

huss commented:
The review by @spearec approves this PR so it is now ready to merge.
